File: credit.java
import java.util.Scanner;

public class Card {
    public static void main(String[] args) {
        System.out.println("Enter Credit card Number");
        Scanner scanner = new Scanner(System.in);
        String card_number = scanner.nextLine();

        int lenght = get_lenght(card_number);
        int[] cardDigit = new int[lenght];
        
        for (int i = 0; i < card_number.length(); i++){
            cardDigit[i] = Character.getNumericValue(card_number.charAt(i));
        }
        if (isValid(cardDigit, lenght)) {
            System.out.println(getType(lenght, card_number));
        }
    }

    public static int get_lenght(String card_number) {
        int lenght = 0;
        for (int i = 0; i < card_number.length(); i++) {
            lenght++;
        }
        return lenght;
    }

    public static boolean isValid(int[] cardDigit, int lenght) {
        int sum = 0;
        for (int i = 0; i < lenght; i++) {
            int digit = cardDigit[i];
            if (lenght % 2 == 0) {
                if (i % 2 == 0) {
                    digit *= 2;

                    if (digit > 9) {
                        digit -= 9;
                    }
                    sum += digit;
                } else {
                    sum += digit;
                }
            } else {
                if (i % 2 != 0) {
                    digit *= 2;

                    if (digit > 9) {
                        digit =- 9;
                    }
                    sum += digit;
                } else {
                    sum += digit;
                }
            }
        }
        return sum % 10 == 0;
    }

    public static String getType(int lenght, String cardNumber) {
        if ((lenght == 16 || lenght == 13) && (40 <= Integer.parseInt(cardNumber.substring(0, 2)) && Integer.parseInt(cardNumber.substring(0, 2)) <= 49)) {
            return "VISA";
        } else if (lenght == 16 && (51 <= Integer.parseInt(cardNumber.substring(0, 2)) && Integer.parseInt(cardNumber.substring(0, 2)) <= 55)) {
            return "MASTERCARD";
        } else if (lenght == 15 && (Integer.parseInt(cardNumber.substring(0, 2)) == 34 || Integer.parseInt(cardNumber.substring(0, 2)) == 37)) {
            return "AMEX";
        } else {
            return "INVALID";
        }
    }
}

File: credit.py
def main():
    creditcard = input("Enter Creditcard number: ")
    if (isValid(creditcard)):
        print(getType(creditcard))

def isValid(creditcard):
    sum = 0
    for i in range(len(creditcard)):
        digit = int(creditcard[i])
        if len(creditcard) % 2 == 0:
            if i % 2 == 0:
                digit *= 2
                if digit > 9:
                    digit -= 9
                sum += digit
            else:
                sum += digit
        else:
            if digit % 2 != 0:
                digit *= 2
                if digit > 9:
                    digit -= 9
                sum += digit
            else:
                sum += digit
    return sum % 10 == 0
    
def getType(creditcard):
    count = len(creditcard)
    n = int(creditcard[:2])
    if (count == 16 or count == 13) and (40 <= n <= 49):
        return "VISA"
    elif count == 16 and (51 <= n <= 55):
        return "MASTERCARD"
    elif count  == 15 and (n == 34 or n == 37):
        return "AMEX"
    else:
        return "INVALID"
if __name__ == "__main__":
    main()